tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -171,6 +171,223 @@ void check_vectors(const T1& t1, const T2& t2) {
+    check_vectors(DataVector(num_pts, 1.0), nine / (one * 9.0));
+    check_vectors(DataVector(num_pts, 9.0), (one * nine) / one);
+
+    // Test powers
+    check_vectors(sqrt(nine), DataVector(num_pts, 3.0));
+    check_vectors(invsqrt(nine), DataVector(num_pts, 1.0 / 3.0));
+    check_vectors(cbrt(eight), DataVector(num_pts, 2.0));
+    check_vectors(invcbrt(eight), DataVector(num_pts, 0.5));
+    check_vectors(DataVector(num_pts, 81.0), pow(nine, 2));
+
+    check_vectors(DataVector(num_pts, 81.0), pow<2>(nine));
+    check_vectors(DataVector(num_pts, 81.0), pow<2>(nine * one));
+    check_vectors(DataVector(num_pts, 1.0 / 81.0),
+                  DataVector(num_pts, 1.0) / pow<2>(nine));
+
+    // Test exp/log
+    check_vectors(DataVector(num_pts, exp(9.0)), exp(nine));
+    check_vectors(DataVector(num_pts, exp2(9.0)), exp2(nine));
+    check_vectors(DataVector(num_pts, pow<9>(10.0)), DataVector(exp10(nine)));
+    check_vectors(DataVector(num_pts, log(9.0)), log(nine));
+    check_vectors(DataVector(num_pts, log2(9.0)), log2(nine));
+    check_vectors(DataVector(num_pts, log10(9.0)), log10(nine));
+
+    // Test trig and other special functions
+    DataVector point_nine(num_pts, 0.9);
+    check_vectors(DataVector(num_pts, sin(9.0)), sin(nine));
+    check_vectors(DataVector(num_pts, cos(9.0)), cos(nine));
+    check_vectors(DataVector(num_pts, tan(9.0)), tan(nine));
+    check_vectors(DataVector(num_pts, asin(0.9)), asin(point_nine));
+    check_vectors(DataVector(num_pts, acos(0.9)), acos(point_nine));
+    check_vectors(DataVector(num_pts, atan(0.9)), atan(point_nine));
+    check_vectors(DataVector(num_pts, atan2(0.9, 9.0)),
+                  atan2(point_nine, nine));
+    check_vectors(DataVector(num_pts, atan2(0.9, 9.0)),
+                  atan2(point_nine * one, nine));
+    check_vectors(DataVector(num_pts, atan2(0.9, 9.0)),
+                  atan2(point_nine, nine * one));
+    check_vectors(DataVector(num_pts, atan2(0.9, 9.0)),
+                  atan2(point_nine * one, nine * one));
+
+    check_vectors(DataVector(num_pts, sinh(9.0)), sinh(nine));
+    check_vectors(DataVector(num_pts, cosh(9.0)), cosh(nine));
+    check_vectors(DataVector(num_pts, tanh(9.0)), tanh(nine));
+    check_vectors(DataVector(num_pts, asinh(9.0)), asinh(nine));
+    check_vectors(DataVector(num_pts, acosh(9.0)), acosh(nine));
+    check_vectors(DataVector(num_pts, atanh(0.9)), atanh(point_nine));
+
+    check_vectors(DataVector(num_pts, erf(0.9)), erf(point_nine));
+    check_vectors(DataVector(num_pts, erfc(0.9)), erfc(point_nine));
+
+    check_vectors(step_function(DataVector{-12.3, 2.0, -4.0, 0.0, 7.0, -8.0}),
+                  DataVector{0.0, 1.0, 0.0, 1.0, 1.0, 0.0});
+  }
+
+  // Test composition of constant expressions with DataVector math member
+  // functions
+  {
+    DataVector x(num_pts, 2.);
+    check_vectors(DataVector(num_pts, 0.82682181043180603), square(sin(x)));
```

Comment:
While you're cleaning this up, these tests could calculate the correct values instead of using hardcoded answers.